# Shopping Mall Customer Segmentation Analysis

## Business Analyst Internship Project

### Project Objective
The objective of this project is to analyze customer demographics, annual income, and spending behavior to identify distinct customer segments.

The insights generated will support business decision-making in:
- Customer targeting
- Marketing optimization
- Customer retention strategies
- Revenue growth initiatives

## Business Problem Statement

Retail businesses serve customers with diverse purchasing behaviors and income levels. 
Without proper segmentation, marketing campaigns become inefficient and fail to maximize customer value.

This project aims to:
- Understand customer characteristics
- Analyze spending patterns
- Identify meaningful customer segments
- Provide actionable business recommendations

## Analytical Approach

The project will follow a structured Business Analytics workflow:

1. Data Understanding
2. Data Cleaning
3. Customer Analysis
4. Customer Segmentation
5. Visualization & Dashboard Development
6. Business Recommendations

## Dataset Overview

The dataset contains customer information collected from a shopping mall.

### Dataset Features
- Customer ID: Unique identifier for each customer
- Gender: Male or Female
- Age: Customer age
- Annual Income (k$): Customer yearly income
- Spending Score (1–100): Indicator of purchasing behavior assigned by the mall

## Part 1: Data Cleaning & Understanding

1. Import the dataset.  
2. Check for missing/null values.  
3. Check data types and clean the dataset.

## Step 1: Import Required Libraries

The following libraries are imported for data manipulation, visualization, and analysis.

In [22]:
# Importing Required Libraries

# Data handling
import pandas as pd
import numpy as np

# Visualization
import matplotlib.pyplot as plt
import seaborn as sns

# Ignore warnings
import warnings
warnings.filterwarnings('ignore')

## Step 2: Load the Dataset

The dataset is loaded into a Pandas DataFrame to enable data exploration and preprocessing.

In [23]:
# Loading Dataset

df = pd.read_csv("../Data/Shopping Mall Customer Segmentation Data .csv")

## Step 3: Preview Dataset

Displaying the first few records helps understand the structure, variables, and overall format of the dataset.

In [24]:
# Previewing Dataset

df.head(3)

,Customer ID,Age,Gender,Annual Income,Spending Score
0,d410ea53-6661-42a9-ad3a-f554b05fd2a7,30,Male,151479,89
1,1770b26f-493f-46b6-837f-4237fb5a314e,58,Female,185088,95
2,e81aa8eb-1767-4b77-87ce-1620dc732c5e,62,Female,70912,76


## Step 4: Dataset Dimensions

Understanding dataset size provides clarity on the number of customers analyzed and available features.

In [25]:
# Understanding Dataset Size

df.shape

(15079, 5)

## Step 5: Data Structure and Data Types

This step identifies:
- Numerical variables suitable for analysis
- Categorical variables for segmentation
- Potential data cleaning requirements

In [26]:
# Understanding Dataset Structure

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 15079 entries, 0 to 15078
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype
---  ------          --------------  -----
 0   Customer ID     15079 non-null  str  
 1   Age             15079 non-null  int64
 2   Gender          15079 non-null  str  
 3   Annual Income   15079 non-null  int64
 4   Spending Score  15079 non-null  int64
dtypes: int64(3), str(2)
memory usage: 589.2 KB


## Step 6: Statistical Summary

Descriptive statistics provide an overview of customer demographics, income distribution, and spending behavior.

In [27]:
# Numerical Summary

df.describe()

,Age,Annual Income,Spending Score
count,15079.000000,15079.000000,15079.000000
mean,54.191591,109742.880562,50.591617
std,21.119207,52249.425866,28.726977
min,18.000000,20022.000000,1.000000
25%,36.000000,64141.000000,26.000000
50%,54.000000,109190.000000,51.000000
75%,72.000000,155008.000000,75.000000
max,90.000000,199974.000000,100.000000


## Step 7: Missing Value Analysis

Missing values can negatively affect analytical results. 

This step verifies data completeness before proceeding to cleaning and analysis.

In [12]:
# Checking Missing Values

df.isnull().sum()

Customer ID       0
Age               0
Gender            0
Annual Income     0
Spending Score    0
dtype: int64

## Step 8: Duplicate Record Check

Duplicate entries may distort customer analysis and segmentation results. 

This step ensures each customer record is unique.

In [13]:
# Checking Duplicate Values

df.duplicated().sum()

np.int64(0)

## Initial Business Observations

### Dataset Overview
The dataset contains information for **15,079 customers** across **5 variables**, representing customer demographics and purchasing behavior within a shopping mall environment.

The available features include:
- Customer ID (unique identifier)
- Age
- Gender
- Annual Income
- Spending Score

The dataset size is sufficiently large to support reliable customer segmentation and behavioral analysis.

---

### Data Quality Assessment
- No missing values were detected across any columns.
- No duplicate customer records were found.
- Data types are appropriately structured:
  - Numerical variables: Age, Annual Income, Spending Score
  - Categorical variables: Gender
  - Identifier variable: Customer ID

This indicates that the dataset is clean and ready for analytical processing without major preprocessing requirements.

---

### Customer Demographic Insights
Initial statistical exploration reveals:

- **Average Customer Age:** ~54 years  
  → Indicates a mature customer base rather than youth-dominated shoppers.

- **Age Range:** 18 to 90 years  
  → Suggests diverse generational customer participation.

---

### Income Characteristics
- **Average Annual Income:** ~109,743
- Income range spans from ~20,000 to ~199,974.

This wide distribution suggests the presence of both low-income and high-income customers, making segmentation highly valuable for targeted marketing strategies.

---

### Spending Behaviour Insights
- **Average Spending Score:** ~50.6 (moderate spending behavior)
- Spending scores range from 1 to 100, indicating strong variability in purchasing engagement.

This variation strongly supports the need for customer segmentation to distinguish high-value and low-engagement customers.

---

### Analytical Direction
Based on preliminary exploration:

- Annual Income and Spending Score are expected to be primary drivers of customer segmentation.
- Demographic variables such as Age and Gender may influence purchasing behavior patterns.
- The dataset is analysis-ready and suitable for advanced segmentation techniques such as clustering.

The next phase will focus on data cleaning validation and preparation for detailed customer analysis.